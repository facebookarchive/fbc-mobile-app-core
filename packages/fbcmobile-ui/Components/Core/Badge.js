/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import * as React from 'react';
import Text from '@fbcmobile/ui/Components/Core/Text';
import {Colors} from '@fbcmobile/ui/Theme';
import {StyleSheet, View} from 'react-native';

export type BadgeVariant = 'primary' | 'secondary';

type Props = {
  +label?: React.Node,
  +icon?: React.Node,
  +variant?: BadgeVariant,
};

const Badge = ({label, variant = 'primary', icon}: Props) => {
  return (
    <View
      style={[
        styles.root,
        variant === 'primary' ? styles.primary : styles.secondary,
      ]}>
      {icon != null ? <View style={styles.icon}>{icon}</View> : null}
      {label != null ? (
        <Text
          style={styles.text}
          variant="h7"
          weight="medium"
          color={variant === 'primary' ? 'light' : 'primary'}>
          {label}
        </Text>
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    height: 30,
    borderRadius: 15,
    backgroundColor: Colors.Blue,
  },
  primary: {
    backgroundColor: Colors.Blue,
  },
  icon: {
    paddingVertical: 3,
    paddingHorizontal: 9,
  },
  text: {
    paddingVertical: 7,
    paddingHorizontal: 10,
  },
  secondary: {
    backgroundColor: Colors.BlueBackground,
  },
});

export default Badge;
