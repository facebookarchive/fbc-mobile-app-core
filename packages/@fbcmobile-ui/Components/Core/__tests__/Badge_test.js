/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import Badge from '../Badge';
import React from 'react';
import renderer from 'react-test-renderer';
import {Icon} from 'react-native-material-ui';

describe('Badge Tests', () => {
  test('Badge renders correctly', () => {
    const tree = renderer.create(<Badge />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Badge renders correctly with label', () => {
    const tree = renderer.create(<Badge label={'testing'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Badge renders correctly with label and variant', () => {
    const tree = renderer
      .create(<Badge label={'testing'} variant={'secondary'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Badge renders correctly with label, variant, and icon', () => {
    const tree = renderer
      .create(
        <Badge
          label={'testing'}
          variant={'secondary'}
          icon={<Icon name={'work'} />}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
