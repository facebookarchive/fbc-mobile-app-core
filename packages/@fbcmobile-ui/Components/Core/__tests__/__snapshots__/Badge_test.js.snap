// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Tests Badge renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#1877F2",
        "borderRadius": 15,
        "height": 30,
      },
      Object {
        "backgroundColor": "#1877F2",
      },
    ]
  }
/>
`;

exports[`Badge Tests Badge renders correctly with label 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#1877F2",
        "borderRadius": 15,
        "height": 30,
      },
      Object {
        "backgroundColor": "#1877F2",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#484848",
            "fontSize": 13,
            "fontWeight": "400",
            "lineHeight": 15,
          },
          Object {
            "color": "white",
          },
          Object {
            "fontFamily": "Roboto-Medium",
            "fontWeight": "500",
          },
          Object {
            "textAlign": "left",
          },
        ],
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 7,
        },
      ]
    }
  >
    testing
  </Text>
</View>
`;

exports[`Badge Tests Badge renders correctly with label and variant 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#1877F2",
        "borderRadius": 15,
        "height": 30,
      },
      Object {
        "backgroundColor": "#e7f3ff",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#484848",
            "fontSize": 13,
            "fontWeight": "400",
            "lineHeight": 15,
          },
          Object {
            "color": "#1877F2",
          },
          Object {
            "fontFamily": "Roboto-Medium",
            "fontWeight": "500",
          },
          Object {
            "textAlign": "left",
          },
        ],
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 7,
        },
      ]
    }
  >
    testing
  </Text>
</View>
`;

exports[`Badge Tests Badge renders correctly with label, variant, and icon 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#1877F2",
        "borderRadius": 15,
        "height": 30,
      },
      Object {
        "backgroundColor": "#e7f3ff",
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 9,
        "paddingVertical": 3,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 24,
          },
          null,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î£¹
    </Text>
  </View>
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#484848",
            "fontSize": 13,
            "fontWeight": "400",
            "lineHeight": 15,
          },
          Object {
            "color": "#1877F2",
          },
          Object {
            "fontFamily": "Roboto-Medium",
            "fontWeight": "500",
          },
          Object {
            "textAlign": "left",
          },
        ],
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 7,
        },
      ]
    }
  >
    testing
  </Text>
</View>
`;
